### PR TITLE
add notification condition to mur

### PR DESCRIPTION
**Pull request by tinakurian**
In order to track the status of the user provisioned notification, we need to add a new Status condition to MasterUserRecord.

This PR adds the conditions for both provisioned notification created and provisioned notification failed. 

Jira: https://issues.redhat.com/browse/CRT-521

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,6 +31,8 @@ require (
 
 // Pinned to kubernetes-1.16.2
 replace (
+    // TODO: tina remove this
+    github.com/codeready-toolchain/api v0.0.0-20200403111848-efd1f59b2e49 => ../api
```
Comment (tinakurian) [inline go.mod:35]: note to myself to remove this

File: pkg/controller/masteruserrecord/masteruserrecord_controller.go [hunk 2]
```diff
@@ -310,6 +310,22 @@ func toBeDisabled() toolchainv1alpha1.Condition {
 	}
 }
 
+func toBeProvisionedNotificationCreated() toolchainv1alpha1.Condition {
```
Comment (tinakurian) [inline pkg/controller/masteruserrecord/masteruserrecord_controller.go:313]: @alexeykazakov not sure if this name is good enough. not a huge fan but i couldn't think of anything better any suggestions?

File: go.mod [hunk 3]
```diff
@@ -31,6 +31,8 @@ require (
 
 // Pinned to kubernetes-1.16.2
 replace (
+    // TODO: tina remove this
+    github.com/codeready-toolchain/api v0.0.0-20200403111848-efd1f59b2e49 => github.com/tinakurian/api v0.0.0-20200505154630-c9659c388982
```
Comment (tinakurian) [inline go.mod:35]: note to myself to remove this

File: pkg/controller/masteruserrecord/masteruserrecord_controller.go [hunk 4]
```diff
@@ -310,6 +310,22 @@ func toBeDisabled() toolchainv1alpha1.Condition {
...
+		Status: corev1.ConditionTrue,
+		Reason: toolchainv1alpha1.MasterUserRecordNotificationCRCreatedReason,
+	}
+}
+
+func toBeProvisionedNotificationFailed() toolchainv1alpha1.Condition {
+	return toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.MasterUserRecordUserProvisionedNotificationCreated,
+		Status: corev1.ConditionFalse,
+		Reason: toolchainv1alpha1.MasterUserRecordNotificationCRCreationFailedReason,
```
Comment (alexeykazakov) [inline pkg/controller/masteruserrecord/masteruserrecord_controller.go:325]: Add a message too. When there is an error then we set the error message as the condition message.

File: pkg/controller/masteruserrecord/masteruserrecord_controller.go (hunk 2 above)
Comment (alexeykazakov) [reply pkg/controller/masteruserrecord/masteruserrecord_controller.go:313]: Yeah... That "toBe" prefix IMO is a bit confusing. But let's keep it for the consistency sake.

Comment (tinakurian) [reply pkg/controller/masteruserrecord/masteruserrecord_controller.go:313]: Yeah agreed, sounds good.

File: pkg/controller/masteruserrecord/masteruserrecord_controller.go (hunk 4 above)
Comment (tinakurian) [reply pkg/controller/masteruserrecord/masteruserrecord_controller.go]: Good idea, added it here: 84fadd766bb49d3a0303a5d271948dbe8e461a5c

**Pull request merged**